ISOFIT 3.5.1 call fails with configuration error if a pre-built LUT is provided.

- EnPT 1.1.0
- ISOFIT 3.5.1

There is a new config validation implemented in ISOFIT 3.5.1 which breaks the implementation within EnPT. After upgrading ISOFIT to v3.5.1, ISOFIT complains about some file paths which cannot be found if a prebuilt LUT is provided:

```python
tests/test_isofit_enmap.py:189: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
enpt/processors/atmospheric_correction/_isofit_enmap.py:869: in _run
    Isofit(
../../../miniforge3/envs/enpt_full_dev/lib/python3.12/site-packages/isofit/core/isofit.py:77: in __init__
    self.config.get_config_errors()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <isofit.configs.configs.Config object at 0x725776c640b0>

    def get_config_errors(self):
        """
        Get configuration option errors by checking the validity of each config section.
        """
        logging.info("Checking config sections for configuration issues")
    
        errors = self.check_config_validity()
        errors.extend(self.check_inter_section_validity())
    
        for e in errors:
            logging.error(e)
    
        if len(errors) > 0:
>           raise AttributeError("Configuration error(s) found.  See log for details.")
E           AttributeError: Configuration error(s) found.  See log for details.

../../../miniforge3/envs/enpt_full_dev/lib/python3.12/site-packages/isofit/configs/configs.py:103: AttributeError

Process finished with exit code 1
```

The used config file looks like this:

```json
{
    "ISOFIT_base": "/home/gfz-fe/miniforge3/envs/enpt_full_dev/lib/python3.12/site-packages",
    "forward_model": {
        "instrument": {
            "parametric_noise_file": "/home/gfz-fe/scheffler/python/EnPT/enpt/resources/EnMAP_Sensor/parametric_noise.txt",
            "integrations": 1,
            "unknowns": {
                "uncorrelated_radiometric_uncertainty": 0.01
            },
            "wavelength_file": "/tmp/tmpotoxzy3j/extracted/backtransformed_l2_spectra_v9/enmap_wavelengths.txt"
        },
        "radiative_transfer": {
            "lut_grid": {
                "AERFRAC_2": [
                    0.05,
                    0.12,
                    0.2,
                    0.3,
                    0.4,
                    0.8
                ],
                "H2OSTR": [
                    0.0,
                    1.0,
                    1.5,
                    2.0,
                    2.7,
                    3.5,
                    5.0
                ],
                "observer_zenith": [
                    0,
                    10.0,
                    20.0,
                    30.0,
                    40.0
                ],
                "solar_zenith": [
                    0,
                    10.0,
                    20.0,
                    35.0,
                    50.0,
                    70.0
                ],
                "relative_azimuth": [
                    0,
                    25.0,
                    50.0,
                    85.0,
                    120.0,
                    155.0,
                    180.0
                ],
                "surface_elevation_km": [
                    0,
                    0.7,
                    2.5,
                    8.0
                ]
            },
            "radiative_transfer_engines": {
                "vswir": {
                    "aerosol_model_file": "/path/to/data/aerosol_model.txt",
                    "aerosol_template_file": "/path/to/data/aerosol_template.json",
                    "earth_sun_distance_file": "/home/gfz-fe/.isofit/data/earth_sun_distance.txt",
                    "emulator_aux_file": "/path/to/srtmnet/sRTMnet_v120_aux.npz",
                    "emulator_file": "/path/to/srtmnet/sRTMnet_v120.h5",
                    "engine_base_dir": "/path/to/6sV2.1",
                    "engine_name": "sRTMnet",
                    "interpolator_base_path": "/path/to/data_out/lut_full/sRTMnet_v120_vi",
                    "irradiance_file": "/home/gfz-fe/.isofit/examples/20151026_SantaMonica/data/prism_optimized_irr.dat",
                    "lut_names": {
                        "AERFRAC_2": null,
                        "H2OSTR": null,
                        "observer_zenith": null,
                        "solar_zenith": null,
                        "relative_azimuth": null,
                        "surface_elevation_km": null
                    },
                    "lut_path": "/tmp/tmpotoxzy3j/EnMAP_LUT_MOD5_ISOFIT_formatted_1nm.nc",
                    "sim_path": "/path/to/data_out/lut_full",
                    "statevector_names": [
                        "AERFRAC_2",
                        "H2OSTR"
                    ],
                    "template_file": "/path/to/data_out/config/20220712T000000Z_modtran_tpl.json"
                }
            },
            "statevector": {
                "AERFRAC_2": {
                    "bounds": [
                        0.05,
                        0.8
                    ],
                    "init": 0.1009,
                    "prior_mean": 0.1009,
                    "prior_sigma": 10.0,
                    "scale": 1
                },
                "H2OSTR": {
                    "bounds": [
                        0,
                        5.0
                    ],
                    "init": 2.5,
                    "prior_mean": 2.5,
                    "prior_sigma": 100.0,
                    "scale": 1
                }
            },
            "unknowns": {
                "H2O_ABSCO": 0.0
            }
        },
        "surface": {
            "select_on_init": true,
            "surface_category": "multicomponent_surface",
            "surface_file": "/tmp/tmpotoxzy3j/extracted/backtransformed_l2_spectra_v9/surface_20221020_EnMAP.mat"
        }
    },
    "implementation": {
        "debug_mode": false,
        "inversion": {
            "windows": [
                [
                    350.0,
                    1360.0
                ],
                [
                    1410,
                    1800.0
                ],
                [
                    1970.0,
                    2500.0
                ]
            ]
        },
        "n_cores": 28,
        "ray_temp_dir": "/tmp/ray"
    },
    "input": {
        "loc_file": "/tmp/tmpotoxzy3j/extracted/backtransformed_l2_spectra_v9/ENMAP01-____L1X-DT0000001702_20220717T110603Z_033_V010400_20231017T193615Z__backtransformed_l2_spectra_loc",
        "measured_radiance_file": "/tmp/tmpotoxzy3j/extracted/backtransformed_l2_spectra_v9/ENMAP01-____L1X-DT0000001702_20220717T110603Z_033_V010400_20231017T193615Z__subX800-810Y370-380",
        "obs_file": "/tmp/tmpotoxzy3j/extracted/backtransformed_l2_spectra_v9/ENMAP01-____L1X-DT0000001702_20220717T110603Z_033_V010400_20231017T193615Z_obs__subX800-810Y370-380_v2"
    },
    "output": {
        "estimated_reflectance_file": "/tmp/tmpotoxzy3j/output/20220717T110603Z_estimated_reflectance.bsq",
        "estimated_state_file": "/tmp/tmpotoxzy3j/output/20220717T110603Z_estimated_state.bsq",
        "posterior_uncertainty_file": "/tmp/tmpotoxzy3j/output/20220717T110603Z_posterior_uncertainty.bsq"
}
```

The following errors are logged:

```bash
2025/09/10 17:14:22:   Checking config sections for configuration issues
2025/09/10 17:14:22:   Config value radiative_transfer->template_file: /path/to/data_out/config/20220712T000000Z_modtran_tpl.json not found
2025/09/10 17:14:22:   Config value radiative_transfer->aerosol_template_file: /path/to/data/aerosol_template.json not found
2025/09/10 17:14:22:   Config value radiative_transfer->aerosol_model_file: /path/to/data/aerosol_model.txt not found
2025/09/10 17:14:22:   Config value radiative_transfer->emulator_aux_file: /path/to/srtmnet/sRTMnet_v120_aux.npz not found
2025/09/10 17:14:22:   Radiative transfer engine file not found on system: /path/to/data/aerosol_model.txt
2025/09/10 17:14:22:   Radiative transfer engine file not found on system: /path/to/data/aerosol_template.json
```

I already tried to replace the values there with None (which I actually thought is the right way), however, ISOFIT still seems to treat the values as file paths:

```bash
2025/09/10 17:17:57:   Checking config sections for configuration issues
2025/09/10 17:17:57:   Config value radiative_transfer->template_file: None not found
2025/09/10 17:17:57:   Config value radiative_transfer->aerosol_template_file: None not found
2025/09/10 17:17:57:   Config value radiative_transfer->aerosol_model_file: None not found
2025/09/10 17:17:57:   Config value radiative_transfer->emulator_aux_file: None not found
2025/09/10 17:17:57:   sRTMnet now requires the emulator_file to be of type .h5.  Please download an updated version from:
 https://zenodo.org/records/10831425
2025/09/10 17:17:57:   Radiative transfer engine file not found on system: None
2025/09/10 17:17:57:   Radiative transfer engine file not found on system: None
```

Any advice on how to set these values now if I want to run ISOFIT based on a pre-built LUT?

--- a/isofit/configs/sections/radiative_transfer_config.py
+++ b/isofit/configs/sections/radiative_transfer_config.py
@@ -200,17 +200,6 @@ class RadiativeTransferEngineConfig(BaseConfigSection):
     def _check_config_validity(self) -> List[str]:
         errors = list()
 
-        # Check that all input files exist
-        for key in self._get_nontype_attributes():
-            value = getattr(self, key)
-            if value and key[-5:] == "_file" and key != "emulator_file":
-                if os.path.isfile(value) is False:
-                    errors.append(
-                        "Config value radiative_transfer->{}: {} not found".format(
-                            key, value
-                        )
-                    )
-
         valid_rt_engines = ["modtran", "6s", "sRTMnet", "KernelFlowsGP"]
         if self.engine_name not in valid_rt_engines:
             errors.append(
@@ -225,47 +214,60 @@ class RadiativeTransferEngineConfig(BaseConfigSection):
                 " available modes: {}".format(self.rt_mode, valid_rt_modes)
             )
 
-        if self.earth_sun_distance_file is None and self.engine_name == "6s":
-            errors.append("6s requires earth_sun_distance_file to be specified")
+        # Only check for missing files when a prebuilt LUT is not provided
+        if not os.path.exists(self.lut_path):
+            # Check that all input files exist
+            for key in self._get_nontype_attributes():
+                value = getattr(self, key)
+                if value and key[-5:] == "_file" and key != "emulator_file":
+                    if os.path.isfile(value) is False:
+                        errors.append(
+                            "Config value radiative_transfer->{}: {} not found".format(
+                                key, value
+                            )
+                        )
 
-        if self.irradiance_file is None and self.engine_name == "6s":
-            errors.append("6s requires irradiance_file to be specified")
+            if self.earth_sun_distance_file is None and self.engine_name == "6s":
+                errors.append("6s requires earth_sun_distance_file to be specified")
 
-        if self.engine_name == "sRTMnet":
-            if self.emulator_file is None:
-                # Fallback to the path specified by the isofit.ini
-                self.emulator_file = env.path("srtmnet", env["srtmnet.file"])
-                if not self.emulator_file.exists():
+            if self.irradiance_file is None and self.engine_name == "6s":
+                errors.append("6s requires irradiance_file to be specified")
+
+            if self.engine_name == "sRTMnet":
+                if self.emulator_file is None:
+                    # Fallback to the path specified by the isofit.ini
+                    self.emulator_file = env.path("srtmnet", env["srtmnet.file"])
+                    if not self.emulator_file.exists():
+                        errors.append(
+                            "The sRTMnet requires an emulator_file to be specified."
+                        )
+
+                if (os.path.splitext(self.emulator_file)[1] != ".h5") and (
+                    os.path.splitext(self.emulator_file)[1] != ".npz"
+                ):
                     errors.append(
-                        "The sRTMnet requires an emulator_file to be specified."
+                        "sRTMnet now requires the emulator_file to be of type .h5 (or .npz for experimental 6c emulator).  "
+                        "Please download an updated version from:\n https://zenodo.org/records/10831425"
                     )
 
-            if (os.path.splitext(self.emulator_file)[1] != ".h5") and (
-                os.path.splitext(self.emulator_file)[1] != ".npz"
-            ):
-                errors.append(
-                    "sRTMnet now requires the emulator_file to be of type .h5 (or .npz for experimental 6c emulator).  "
-                    "Please download an updated version from:\n https://zenodo.org/records/10831425"
-                )
+                if self.emulator_aux_file is None:
+                    # Fallback to the path specified by the isofit.ini
+                    self.emulator_aux_file = env.path("srtmnet", env["srtmnet.aux"])
+                    if not self.emulator_aux_file.exists():
+                        errors.append(
+                            "The sRTMnet requires an emulator_aux_file to be specified."
+                        )
 
-            if self.emulator_aux_file is None:
-                # Fallback to the path specified by the isofit.ini
-                self.emulator_aux_file = env.path("srtmnet", env["srtmnet.aux"])
-                if not self.emulator_aux_file.exists():
+            files = [
+                self.obs_file,
+                self.aerosol_model_file,
+                self.aerosol_template_file,
+            ]
+            for file in files:
+                if file is not None and not os.path.isfile(file):
                     errors.append(
-                        "The sRTMnet requires an emulator_aux_file to be specified."
+                        f"Radiative transfer engine file not found on system: {file}"
                     )
-
-        files = [
-            self.obs_file,
-            self.aerosol_model_file,
-            self.aerosol_template_file,
-        ]
-        for file in files:
-            if file is not None and not os.path.isfile(file):
-                errors.append(
-                    f"Radiative transfer engine file not found on system: {file}"
-                )
 
         if isinstance(self.lut_complevel, int) and self.lut_complevel < 1:
             errors.append("The LUT complevel must be and int greater than 0")

